sql/moz-fx-data-marketing-prod/google_search_console/search_impressions_by_page/view.sql

```diff
@@ -1,48 +1,135 @@
+    user_country_code,
+    device_type,
+    impressions,
+    clicks,
+    average_position
+  FROM
+    `moz-fx-data-marketing-prod.google_search_console_derived.search_impressions_by_page_v1`
+  WHERE
+    CASE
+      WHEN site_domain_name IN (
+          'addons.mozilla.org',
+          'blog.mozilla.org',
+          'getpocket.com',
+          'support.mozilla.org',
+          'www.mozilla.org'
+        )
+        THEN `date` < '2023-08-01'
+      WHEN site_domain_name = 'developer.mozilla.org'
+        THEN `date` < '2024-04-10'
+      ELSE FALSE
+    END
+  UNION ALL
+  SELECT
+    `date`,
+    site_url,
+    site_domain_name,
+    page_url,
+    page_domain_name,
+    page_path,
+    localized_site_code,
+    localized_site_language_code,
+    localized_site_country_code,
+    query,
+    is_anonymized,
+    has_good_page_experience,
+    search_type,
+    search_appearance,
+    user_country_code,
+    device_type,
+    impressions,
+    clicks,
+    average_position
+  FROM
+    `moz-fx-data-marketing-prod.google_search_console_derived.search_impressions_by_page_v2`
+  WHERE
+    CASE
+      WHEN site_domain_name IN (
+          'addons.mozilla.org',
+          'blog.mozilla.org',
+          'getpocket.com',
+          'support.mozilla.org',
+          'www.mozilla.org'
+        )
+        THEN `date` >= '2023-08-01'
+      ELSE TRUE
+    END
+)
 SELECT
-  `date`,
-  site_url,
-  site_domain_name,
-  page_url,
-  page_domain_name,
-  page_path,
-  page_path_segment_1,
-  query,
-  FALSE AS is_anonymized,
-  CAST(NULL AS BOOLEAN) AS has_good_page_experience,
-  search_type,
-  CAST(NULL AS STRING) AS search_appearance,
-  country_code,
-  device_type,
-  impressions,
-  clicks,
-  average_position
+  search_impressions.date,
+  search_impressions.site_url,
+  search_impressions.site_domain_name,
+  search_impressions.page_url,
+  search_impressions.page_domain_name,
+  search_impressions.page_path,
+  search_impressions.localized_site_code,
+  CONCAT(
+    COALESCE(localized_site_language.name, search_impressions.localized_site_language_code),
+    COALESCE(
+      CONCAT(
+        ' - ',
+        COALESCE(localized_site_country.name, search_impressions.localized_site_country_code)
+      ),
+      ''
+    )
+  ) AS localized_site,
+  search_impressions.localized_site_language_code,
+  COALESCE(
+    localized_site_language.name,
+    search_impressions.localized_site_language_code
+  ) AS localized_site_language,
+  search_impressions.localized_site_country_code,
+  COALESCE(
+    localized_site_country.name,
+    search_impressions.localized_site_country_code
+  ) AS localized_site_country,
+  search_impressions.query,
+  -- Discover and Google News search impressions never have `query` values.
+  IF(
+    search_impressions.search_type IN ('Discover', 'Google News'),
+    NULL,
+    mozfun.google_search_console.classify_site_query(
+      search_impressions.site_domain_name,
+      search_impressions.query
+    )
+  ) AS query_type,
```

Comment:
What's the difference in the meaning of a null query type vs the `Anonymized` the udf returns for null queries?

Reply:
For Discover and Google News searches there is never a query value, so a query type isn't applicable (I suppose I could have had the query type be something like `None` or `N/A` in those cases, but I felt having it be null was more appropriate/idiomatic).

For all other search types there should be a query value unless the search was explicitly anonymized.